modules/osdk-run-proxy.adoc

```diff
@@ -0,0 +1,123 @@
+// Module included in the following assemblies:
+//
+// * operators/operator_sdk/golang/osdk-golang-tutorial.adoc
+// * operators/operator_sdk/ansible/osdk-ansible-tutorial.adoc
+// * operators/operator_sdk/helm/osdk-helm-tutorial.adoc
+
+ifeval::["{context}" == "osdk-golang-tutorial"]
+:golang:
+:app-proper: Memcached
+:app: memcached
+:group: cache
+endif::[]
+ifeval::["{context}" == "osdk-ansible-tutorial"]
+:ansible:
+:app-proper: Memcached
+:app: memcached
+:group: cache
+endif::[]
+ifeval::["{context}" == "osdk-helm-tutorial"]
+:helm:
+:app-proper: Nginx
+:app: nginx
+:group: demo
+endif::[]
+
+[id="osdk-run-proxy_{context}"]
+
+= Proxy support
+
+To support proxied clusters, your Operator must inspect the environment for the following standard proxy variables and pass the values to Operands:
+
+- `HTTPS_PROXY`
+- `HTTPS_PROXY`
+- `NO_PROXY`
+
+
+ifdef::golang[]
+Add the `proxy.ReadProxyVarsFromEnv` helper function to the reconcile loop in the `controllers/memcached_controller.go` file and append the results to the Operand environments:
+
+[source,golang,subs="attributes+"]
+----
+for i, container := range dep.Spec.Template.Spec.Containers {
+		dep.Spec.Template.Spec.Containers[i].Env = append(container.Env, proxy.ReadProxyVarsFromEnv()...)
+}
+----
+
+endif::[]
+
+ifdef::helm[]
+To include the overrides based on an environment variable, modify the `watches.yaml` file to include:
+
+[source,yaml,subs="attributes+"]
+----
+- group: demo.example.com
+  version: v1alpha1
+  kind: Nginx
+  chart: helm-charts/nginx
+  overrideValues:
+    proxy.http: $HTTP_PROXY
+#+kubebuilder:scaffold:watch
+----
+
+Add the `proxy.http` value in the `helmcharts/nginx/Values.yaml` file:
+
+[source,yaml,subs="attributes+"]
+----
+proxy:
+  http: ""
+  https: ""
+  no_proxy: ""
+----
+
+To make sure the chart template supports using the variables, edit the chart template in the `helm-charts/nginx/templates/deployment.yaml` file to contain the following:
+
+[source,yaml,subs="attributes+"]
+----
+containers:
+  - name: {{ .Chart.Name }}
+    securityContext:
+      - toYaml {{ .Values.securityContext | nindent 12 }}
+    image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+    imagePullPolicy: {{ .Values.image.pullPolicy }}
+    env:
+      - name: http_proxy
+        value: "{{ .Values.proxy.http }}"
+----
+
+endif::[]
+
+Set the environment variable on the Operator deployment by adding the following to the `config/manager/manager.yaml` file:
+
+[source,yaml,subs="attributes+"]
+----
+containers:
+ - args:
+   - --leader-elect
+   - --leader-election-id=ansible-proxy-demo
+   image: controller:latest
+   name: manager
+   env:
+     - name: "HTTP_PROXY"
+       value: "http_proxy_test"
+----
+
```

Comment:
We should add the example that add the env to the deployment. such as:
task/main.yaml
---
- name: start memcached
  community.kubernetes.k8s:
    definition:
      kind: Deployment
      apiVersion: apps/v1
      metadata:
        name: '{{ ansible_operator_meta.name }}-memcached'
        namespace: '{{ ansible_operator_meta.namespace }}'
      spec:
        replicas: "{{size}}"
        selector:
          matchLabels:
            app: memcached
        template:
          metadata:
            labels:
              app: memcached
          spec:
            containers:
            - name: memcached
              command:
              - memcached
              - -m=64
              - -o
              - modern
              - -v
              image: "docker.io/memcached:1.4.36-alpine"
              ports:
                - containerPort: 11211
              env:
                 - Name: HTTP_PROXY: 
                   Value: "{{ lookup('env', 'HTTP_PROXY') | default('', True) }}"
                 - Name: http_proxy: 
                   Value: "{{ lookup('env', 'HTTP_PROXY') | default('', True) }}"

Reply:
@emmajiafan I have made the requested changes to the [Ansible tutorial](https://deploy-preview-36738--osdocs.netlify.app/openshift-enterprise/latest/operators/operator_sdk/ansible/osdk-ansible-tutorial.html#osdk-run-proxy_osdk-ansible-tutorialr). Would you PTAL?

Reply:
@michaelryanpeter  /lgtm